packages/website/ts/components/relayer_index/relayer_index.tsx

```diff
@@ -25,94 +32,65 @@ const styles: Styles = {
-        name: 'Radar Relay',
-        marketShare: 0.5,
-        topTokens,
-    },
-    {
-        id: '2',
-        headerUrl: '/images/og_image.png',
-        name: 'Paradex',
-        marketShare: 0.5,
-        topTokens,
-    },
-    {
-        id: '3',
-        headerUrl: '/images/og_image.png',
-        name: 'yo',
-        marketShare: 0.5,
-        topTokens,
-    },
-    {
-        id: '4',
-        headerUrl: '/images/og_image.png',
-        name: 'test',
-        marketShare: 0.5,
-        topTokens,
-    },
-    {
-        id: '5',
-        headerUrl: '/images/og_image.png',
-        name: 'blahg',
-        marketShare: 0.5,
-        topTokens,
-    },
-    {
-        id: '6',
-        headerUrl: '/images/og_image.png',
-        name: 'hello',
-        marketShare: 0.5,
-        topTokens,
-    },
-];
-
 const CELL_HEIGHT = 260;
 const NUMBER_OF_COLUMNS = 4;
 const GRID_PADDING = 16;
 
-export const RelayerIndex: React.StatelessComponent<RelayerIndexProps> = (props: RelayerIndexProps) => {
-    return (
-        <div style={styles.root}>
-            <GridList cellHeight={CELL_HEIGHT} cols={NUMBER_OF_COLUMNS} padding={GRID_PADDING} style={styles.gridList}>
-                {relayerInfos.map((relayerInfo: RelayerInfo) => (
-                    <RelayerGridTile key={relayerInfo.id} relayerInfo={relayerInfo} networkId={props.networkId} />
-                ))}
-            </GridList>
-        </div>
-    );
-};
+export class RelayerIndex extends React.Component<RelayerIndexProps, RelayerIndexState> {
+    private _isUnmounted: boolean;
+    constructor(props: RelayerIndexProps) {
+        super(props);
+        this._isUnmounted = false;
+        this.state = {
+            relayerInfos: undefined,
+            error: undefined,
+        };
+    }
+    public componentWillMount() {
+        // tslint:disable-next-line:no-floating-promises
+        this._fetchRelayerInfosAsync();
+    }
+    public componentWillUnmount() {
+        this._isUnmounted = true;
+    }
+    public render() {
+        // TODO: loading and error states with a scrolling container
+        const readyToRender = _.isUndefined(this.state.error) && !_.isUndefined(this.state.relayerInfos);
+        return (
+            readyToRender && (
+                <div style={styles.root}>
+                    <GridList
+                        cellHeight={CELL_HEIGHT}
+                        cols={NUMBER_OF_COLUMNS}
+                        padding={GRID_PADDING}
+                        style={styles.gridList}
+                    >
+                        {this.state.relayerInfos.map((relayerInfo: WebsiteBackendRelayerInfo) => (
+                            <RelayerGridTile
+                                key={relayerInfo.id}
+                                relayerInfo={relayerInfo}
+                                networkId={this.props.networkId}
+                            />
+                        ))}
+                    </GridList>
+                </div>
```

Comment:
We could add a spinner here for the loading state. For the error state we should also show the user a helpful error message. Perhaps to refresh the page?

Even if you plan to do this in a later PR before make this a two part conditional with a TODO for the `!readyToRender` case?